### ISSUE TITLE
Could not find action or lane 'book' when using imported Fastfile

### ISSUE THREAD

**Issue by AnthonyMDev** [fastlane/fastlane#6551]
### New Issue Checklist
- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
### Issue Description

I have a `Fastfile` that is importing another `Fastfile`. I get the following error when I try to do any action.
##### Complete output when running fastlane, including the stack trace and command used

```
fastlane lanes --verbose
DEBUG [2016-12-21 13:28:13.83]: Checking if there are any plugins that should be loaded...
DEBUG [2016-12-21 13:28:13.84]: /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/runner.rb:140:in `trigger_action_by_name'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
../../PushPodFastfile:1:in `parsing_binding'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `eval'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `block in parse'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `chdir'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `parse'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:209:in `import'
Fastfile:2:in `parsing_binding'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `eval'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `block in parse'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `chdir'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `parse'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:33:in `initialize'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/lane_list.rb:11:in `new'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/lane_list.rb:11:in `generate'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/lane_list.rb:5:in `output'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/commands_generator.rb:141:in `block (2 levels) in run'
/usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/command.rb:178:in `call'
/usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/command.rb:153:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/runner.rb:444:in `run_active_command'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
/usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/delegates.rb:15:in `run!'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/commands_generator.rb:293:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/commands_generator.rb:36:in `start'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/cli_tools_distributor.rb:59:in `take_off'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/bin/fastlane:5:in `<top (required)>'
/usr/local/bin/fastlane:23:in `load'
/usr/local/bin/fastlane:23:in `<main>'
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane_core/lib/fastlane_core/ui/interface.rb:145:in `user_error!': [!] Could not find action or lane 'book'. Check out the README for more details: https://github.com/fastlane/fastlane/tree/master/fastlane (FastlaneCore::Interface::FastlaneError)
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane_core/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/runner.rb:154:in `rescue in trigger_action_by_name'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/runner.rb:140:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	from ../../PushPodFastfile:1:in `parsing_binding'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `eval'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `block in parse'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `chdir'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `parse'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:209:in `import'
	from Fastfile:2:in `parsing_binding'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `eval'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:54:in `block in parse'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `chdir'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:43:in `parse'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/fast_file.rb:33:in `initialize'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/lane_list.rb:11:in `new'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/lane_list.rb:11:in `generate'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/lane_list.rb:5:in `output'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/commands_generator.rb:141:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/runner.rb:444:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
	from /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/commands_generator.rb:293:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/commands_generator.rb:36:in `start'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/fastlane/lib/fastlane/cli_tools_distributor.rb:59:in `take_off'
	from /usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.2.0/bin/fastlane:5:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```
### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:

``` ruby
################
# Configuration:
################

ENV["SLACK_URL"] ||= "https://hooks.slack.com/services/PRIVATE_SLACK_URL"
default_platform :ios
#update_fastlane
fastlane_version "2.2.0"

################
# Helpers:
################

def bump_podspec(pod_name, version)
    podspec_path = "#{pod_name}.podspec"
    version_bump_podspec(path: podspec_path, version_number: version)
    git_commit(path: podspec_path, message: "Updated podspec to version #{version}.")
    add_git_tag(tag: version)
    push_to_git_remote
end

def notify_slack(message)
    slack(channel: "development",
          default_payloads: [],
          message: message)
end

################
# Lanes:
################

platform :ios do
  before_all do

  end

    # test
    desc "Runs all the tests"
    lane :test do
        scan
    end

    desc "Push a new pod version"
    lane :push_pod do |options|
        ensure_git_status_clean
        pod_name = POD_NAME
        version = options[:version]

        bump_podspec(pod_name, version)

        pod_push(
            path: "#{pod_name}.podspec",
            repo: "Zorts",
            sources: ["MY_PRIVATE_SOURCE", "https://github.com/CocoaPods/Specs"]
        )
        rocket

        current_branch = git_branch

        if current_branch != "develop"
            push_to_git_remote(
                remote_branch: "develop",
                tags: false
            )
        end

        if current_branch != "master"
            push_to_git_remote(
                remote_branch: "master",
                tags: false
            )
        end

        notify_slack("Pushed #{pod_name} #{version} to the Zorts Spec Repository. :rocket:")
        notification(subtitle: "Fastlane - Success" , message: "#{pod_name} #{version} pushed.")
    end

    ################
    # Success/Error:
    ################

    after_each do |lane|
        clean_build_artifacts
    end

    after_all do |lane|
        reset_git_repo
    end

    error do |lane, exception|
        reset_git_repo
        notification(subtitle: "Fastlane #{lane} failed.", message: "")
    end
end
```
### Environment

fastlane version (run `fastlane -v`): 1.105.2

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? No

What Xcode version do you use (e.g. 7.3, 8.0) 8.0

**Comment by TKBurner** [comment]
@AnthonyMDev Thanks for reaching out. Can you provide your other fastfile? Has this worked previously or is this a new setup? Thanks!

**Comment by AnthonyMDev** [comment]
This is a new setup.

The second `Fastfile` is just:

````ruby
ENV["POD_NAME"] = "ZRAPI"
import './../../PushPodFastfile'
```

Not sure if I'm using the environment variables properly there yet. I was trying to test how that worked when I hit this issue, which has precluded me moving on.

Thanks for looking at this!

**Comment by TKBurner** [comment]
@AnthonyMDev Check out the steps here, https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Advanced.md#importing-another-fastfile

**Comment by AnthonyMDev** [comment]
@TKBurner Thanks. I had already looked at that in the first place. That looks to be what I'm doing though and it's not working. (Hence the issue submission :) )

**Comment by TKBurner** [comment]
@AnthonyMDev Is your PushPodFastfile nested that many layers below the directory that your initial Fastfile is? Or, are you giving the path from root?

**Comment by AnthonyMDev** [comment]
It's there. Not giving path to root. If I change the path I get a different error that it can't find the imported fast file so I'm pretty certain that's right.

**Comment by TKBurner** [comment]
@AnthonyMDev Thanks for following up. If you run lane on the file directly does it have the proper output? I want to isolate where this issue is.

**Comment by AnthonyMDev** [comment]
Yes, running the original fast file's `push_pod` lane works just fine. The issue only occurred when I tried to import the fast file in another fast file. I want to use this fast file in multiple projects so I need to import it.

The error is saying `Could not find action or lane 'book'.` but I have no reference in either of my fast files to the word `book` anywhere, so I'm assuming this is an internal bug in the fastlane software.

**Comment by TKBurner** [comment]
@AnthonyMDev I looped in some other team members to get some input here. We'll work with you on this thread. Thanks!

**Comment by mpirri** [comment]
@AnthonyMDev Sorry for the delay in getting back to you about this.

I set up a simple folder hierarchy that, from what I can tell, should match your directory structure.  Here's what I have:
![image](https://cloud.githubusercontent.com/assets/1190389/19945044/aaf74a98-a114-11e6-9b7e-483f24d0abdd.png)

`PushPodFastfile` contains the 94 lines from your initial issue posting
`Fastfile` contains the 2 lines from your first comment

When I change directory into `directory` and run `fastlane lanes`, I get:
```
[15:51:28]: ------------------------------
[15:51:28]: --- Step: default_platform ---
[15:51:28]: ------------------------------
[15:51:28]: -------------------------------------------------
[15:51:28]: --- Step: Verifying required fastlane version ---
[15:51:28]: -------------------------------------------------
[15:51:28]: Your fastlane version 1.106.2 matches the minimum requirement of 1.105.2  ✅

--------- ios---------
----- fastlane ios test
Runs all the tests

----- fastlane ios push_pod
Push a new pod version

Execute using `fastlane [lane_name]`
```

Mind you, I am running _fastlane_ version 1.106.2, you might want to upgrade to see if the new version has fixed the problem.

Does this sound like an appropriate analogue for what you are doing?

Any chance you can you try a similar experiment? Just set up a minimal project structure and try doing `fastlane lanes`? I'm curious to know if there is something wrong with _fastlane_ itself or if something else about your project is causing difficulty here.

**Comment by AnthonyMDev** [comment]
@mpirri Sorry I didn't get back to you for a while. I've been on vacation!

I've set up a simple folder hierarchy just like you've described, and I'm still getting the same error.

**Comment by mpirri** [comment]
@AnthonyMDev Did you try updating _fastlane_ to the latest version? The fact that you are seeing different behavior here seems to indicate that something is amiss with either your _fastlane_ install or a global configuration option (maybe an environment variable or a configuration file somewhere).

**Comment by AnthonyMDev** [comment]
@mpirri Thanks for all of your advice so far. I have updated `fastlane` multiple times and seen this same issue.

If there are other env variables or configurations that you would like me to look at please let me know.

**Comment by AnthonyMDev** [comment]
I'm now using fastlane `2.2.0` and this still isn't working. It seems to be an issue with the `parsing_binding` method in `fast_file.rb:146`.

**Comment by AnthonyMDev** [comment]
I just figured out what was going on here. The `Fastfile` that I was importing as an alias. I actually am storing my shared `Fastfile` in Dropbox and using a symlink to store it at the path I am importing. When I moved the actual file to that path, it worked just fine.

So it seems the issue is importing a `Fastfile` that is symlinked. Far smaller issue, but something that may want to be addressed.

@mpirri I'm not sure if you want to close this issue or keep it open for dealing with the symlink issue.
